core/src/test/java/com/linecorp/armeria/common/HttpResponseTest.java

```diff
@@ -150,4 +154,56 @@ void ofRedirectParamsShouldNotBeNull() {
             null))
             .isInstanceOf(NullPointerException.class).hasMessageContaining("args");
     }
+
+    @Test
+    void delayedHttpResponseWithAggregatedHttpResponseUsingCurrentEventLoopOrCommonPools() {
+        final AggregatedHttpResponse aggregatedHttpResponse = AggregatedHttpResponse.of(HttpStatus.OK);
+        final HttpResponse res = HttpResponse.delayed(aggregatedHttpResponse, Duration.ofSeconds(1));
+        assertThat(res.aggregate().join().status()).isEqualTo(HttpStatus.OK);
```

Comment:
Could we check if 1 second has actually passed?
Perhaps use Guava's `StopWatch`?

Reply:
Thank you 😄 Let me apply it !